src/frontend/src/handler/create_table.rs

```diff
@@ -1320,12 +1349,334 @@ pub async fn handle_create_table(
         serde_json::to_string_pretty(&graph).unwrap()
     );
 
+    // Handle engine
+    match engine {
+        Engine::Hummock => {
+            let catalog_writer = session.catalog_writer()?;
+            catalog_writer
+                .create_table(source, table, graph, job_type)
+                .await?;
+        }
+        Engine::Iceberg => {
+            create_iceberg_engine_table(
+                session,
+                handler_args,
+                source,
+                table,
+                graph,
+                job_type,
+                column_defs,
+                constraints,
+                table_name,
+            )
+            .await?;
+        }
+    }
+
+    Ok(PgResponse::empty_result(StatementType::CREATE_TABLE))
+}
+
+#[allow(clippy::too_many_arguments)]
+pub async fn create_iceberg_engine_table(
+    session: Arc<SessionImpl>,
+    handler_args: HandlerArgs,
+    source: Option<PbSource>,
+    table: PbTable,
+    graph: StreamFragmentGraph,
+    job_type: TableJobType,
+    column_defs: Vec<ColumnDef>,
+    constraints: Vec<TableConstraint>,
+    table_name: ObjectName,
+) -> Result<()> {
+    // 1. fetch iceberg engine options from the meta node.
+    // 2. create a hummock table
+    // 3. create an iceberg sink
+    // 4. create an iceberg source
+
+    let meta_client = session.env().meta_client();
+    let system_params = meta_client.get_system_params().await?;
+    let state_store_endpoint = system_params.state_store().to_string();
+    let data_directory = system_params.data_directory().to_string();
+    let meta_store_endpoint = meta_client.get_meta_store_endpoint().await?;
+
+    let s3_region = if let Ok(s3_region) = std::env::var("AWS_REGION") {
+        s3_region
+    } else {
+        bail!("To create an iceberg engine table, AWS_REGION needed to be set");
+    };
```

Comment:
Why?

Reply:
https://github.com/risingwavelabs/risingwave/pull/19577#discussion_r1867268505

Reply:
I think we can check this one for s3 only. For minio, we can always pass a default s3 region.